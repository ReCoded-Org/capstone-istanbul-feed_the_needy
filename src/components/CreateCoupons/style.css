.ant-btn.ant-btn-primary.ant-btn-round.ant-btn-lg.ant-btn-background-ghost.couponPrices {
  /*priceButtons*/
  font-size: large;
  color: black;
  width: 6rem;
}

.ant-btn.ant-btn-primary.ant-btn-round.ant-btn-lg.couponPrices {
  /*selectedPriceButton*/
  width: 6rem;
  background-color: #e17007;
  font-size: large;
}

.ant-avatar.ant-avatar-circle.ant-avatar-icon.couponInfo {
  /*infoAvatar*/
  background-color: white;
  color: black;
}

.ant-btn.ant-btn-primary.ant-btn-lg.couponCreateButton {
  /*createButton*/
  width: 10rem;
  background-color: #e17007;
  color: white;
}

.createCoupons {
  padding: 1rem 0;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.couponInfoWrapper {
  padding-bottom: 1rem;
  text-align: center;
}

.createCouponsTitle {
  padding-bottom: 1rem;
}
