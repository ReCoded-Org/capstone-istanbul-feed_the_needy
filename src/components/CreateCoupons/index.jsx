import React, { useState } from "react";
import "./style.css";
import { Typography, Button, Row, Col, Avatar, message, Divider } from "antd";
import { InfoCircleOutlined } from "@ant-design/icons";
import { useTranslation } from "react-i18next";
import firebase from "../../firebaseConfig";

const db = firebase.firestore();
const timestamp = firebase.firestore.FieldValue.serverTimestamp;
const { Title, Text } = Typography;

const CreateCoupons = ({ isTesting }) => {
  const { t } = useTranslation();
  const [count, setCount] = useState(1);
  const [newCoupons, setNewCoupons] = useState({
    compName: "",
    amounts: {
      25: false,
      50: false,
      75: false,
      100: false,
      150: false,
      200: false,
    },
  });
  const [ghosted, setGhosted] = useState({
    // ghost property in antd button removes the styling from it
    "25₺": true,
    "50₺": true,
    "75₺": true,
    "100₺": true,
    "150₺": true,
    "200₺": true,
  });

  const postCoupons = () => {
    // db.collection("restaurants").where("uid", "==", "") get the uid from the auth()
    db.collection("coupons")
      .add({
        compName: newCoupons.compName,
        amounts: { ...newCoupons.amounts },
        createdAt: timestamp(),
        img:
          "https://image.freepik.com/free-photo/wooden-board-empty-table-top-blurred-background_1253-1584.jpg",
      })
      .then((docRef) =>
        db.collection("coupons").doc(docRef.id).update({
          id: docRef.id,
        })
      )
      .then(() => message.success(t("createCoupons.message.success")))
      .catch(() => message.warning(t("createCoupons.message.warning")));
  };

  const selectCoupon = (e) => {
    const key = e.target.innerText;
    const slicedKey = key.slice(0, key.length - 1);
    setGhosted({
      ...ghosted,
      [key]: !ghosted[key],
    });

    setNewCoupons({
      ...newCoupons,
      compName: "Arby's",
      amounts: {
        ...newCoupons.amounts,
        [slicedKey]: !newCoupons.amounts[slicedKey],
      },
    });

    if (ghosted[key] === false) {
      setCount((previous) => previous - 1);
    } else {
      setCount((previous) => previous + 1);
    }
  };

  const handleSubmit = () => {
    postCoupons();
    setGhosted({
      "25₺": true,
      "50₺": true,
      "75₺": true,
      "100₺": true,
      "150₺": true,
      "200₺": true,
    });
    setCount(1);
  };

  const firstRowList = [];
  const secondRowList = [];
  const ghostList = [
    ghosted["25₺"],
    ghosted["50₺"],
    ghosted["75₺"],
    ghosted["100₺"],
    ghosted["150₺"],
    ghosted["200₺"],
  ];
  const coupons = ["25₺", "50₺", "75₺", "100₺", "150₺", "200₺"];

  for (let i = 0; i < ghostList.length / 2; i += 1) {
    firstRowList.push({ ghost: ghostList[i], text: coupons[i] });
  }
  for (let i = ghostList.length / 2; i < ghostList.length; i += 1) {
    secondRowList.push({ ghost: ghostList[i], text: coupons[i] });
  }

  const MAXSELECTABLECOUPON = 4;

  return (
    <div className="createCoupons">
      <Divider className="createCouponsTitle">
        <Title level={3} type="primary">
          {t("createCoupons.title")}
        </Title>
      </Divider>
      <Row gutter={[16, 16]}>
        {firstRowList.map((row) => (
          <Col key={row.text} span={8}>
            {isTesting ? (
              <button type="button">{row.text}</button>
            ) : (
              <Button
                className="couponPrices"
                onClick={selectCoupon}
                size="large"
                shape="round"
                ghost={row.ghost} // default is true and when selected false
                type="primary"
                disabled={count > MAXSELECTABLECOUPON && row.ghost}
              >
                <b>{row.text}</b>
              </Button>
            )}
          </Col>
        ))}
      </Row>
      <Row gutter={[16, 16]}>
        {secondRowList.map((row) => (
          <Col key={row.text} span={8}>
            {isTesting ? (
              <button type="button">{row.text}</button>
            ) : (
              <Button
                className="couponPrices"
                onClick={selectCoupon}
                size="large"
                shape="round"
                ghost={row.ghost}
                type="primary"
                disabled={count > MAXSELECTABLECOUPON && row.ghost}
              >
                <b>{row.text}</b>
              </Button>
            )}
          </Col>
        ))}
      </Row>
      <div className="couponInfoWrapper">
        <Avatar
          className="couponInfo"
          size={{ xs: 24, sm: 32, md: 40, lg: 64, xl: 80, xxl: 100 }}
          icon={<InfoCircleOutlined />}
        />
        <Text>
          <b>{t("createCoupons.text")}</b>
        </Text>
      </div>
      {isTesting ? (
        <button type="button">25₺</button>
      ) : (
        <Button
          onClick={handleSubmit}
          className="couponCreateButton"
          type="primary"
          size="large"
          disabled={count <= MAXSELECTABLECOUPON}
        >
          {t("createCoupons.createButton")}
        </Button>
      )}
    </div>
  );
};

export default CreateCoupons;
