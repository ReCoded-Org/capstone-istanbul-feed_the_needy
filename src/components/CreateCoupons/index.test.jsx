import React from "react";
import renderer from "react-test-renderer";
import CreateCoupons from "./index";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

// Fix for TypeError: window.matchMedia is not a function
Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

it("CreateCoupons renders correctly", () => {
  const tree = renderer.create(<CreateCoupons isTesting />).toJSON();
  expect(tree).toMatchSnapshot();
});
