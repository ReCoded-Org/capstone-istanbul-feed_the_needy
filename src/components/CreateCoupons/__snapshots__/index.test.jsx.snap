// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCoupons renders correctly 1`] = `
<div
  className="coupons"
>
  <div
    className="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-center createCouponsTitle"
    role="separator"
  >
    <span
      className="ant-divider-inner-text"
    >
      <h3
        className="ant-typography ant-typography-primary"
        style={
          Object {
            "WebkitLineClamp": null,
          }
        }
      >
        createCoupons.title
      </h3>
    </span>
  </div>
  <div
    className="ant-row"
    style={
      Object {
        "marginBottom": 8,
        "marginLeft": -8,
        "marginRight": -8,
        "marginTop": -8,
      }
    }
  >
    <div
      className="ant-col ant-col-8"
      style={
        Object {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        }
      }
    >
      <button
        type="button"
      >
        25₺
      </button>
    </div>
    <div
      className="ant-col ant-col-8"
      style={
        Object {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        }
      }
    >
      <button
        type="button"
      >
        50₺
      </button>
    </div>
    <div
      className="ant-col ant-col-8"
      style={
        Object {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        }
      }
    >
      <button
        type="button"
      >
        75₺
      </button>
    </div>
  </div>
  <div
    className="ant-row"
    style={
      Object {
        "marginBottom": 8,
        "marginLeft": -8,
        "marginRight": -8,
        "marginTop": -8,
      }
    }
  >
    <div
      className="ant-col ant-col-8"
      style={
        Object {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        }
      }
    >
      <button
        type="button"
      >
        100₺
      </button>
    </div>
    <div
      className="ant-col ant-col-8"
      style={
        Object {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        }
      }
    >
      <button
        type="button"
      >
        150₺
      </button>
    </div>
    <div
      className="ant-col ant-col-8"
      style={
        Object {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        }
      }
    >
      <button
        type="button"
      >
        200₺
      </button>
    </div>
  </div>
  <div
    className="couponInfoWrapper"
  >
    <span
      className="ant-avatar ant-avatar-circle ant-avatar-icon couponInfo"
      style={Object {}}
    >
      <span
        aria-label="info-circle"
        className="anticon anticon-info-circle"
        role="img"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="info-circle"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
          />
          <path
            d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
          />
        </svg>
      </span>
    </span>
    <span
      className="ant-typography"
      style={
        Object {
          "WebkitLineClamp": null,
        }
      }
    >
      <b>
        createCoupons.text
      </b>
    </span>
  </div>
  <button
    type="button"
  >
    25₺
  </button>
</div>
`;
